scene/gui/rich_text_label.cpp

```diff
@@ -1596,19 +1596,33 @@ void RichTextLabel::add_text(const String &p_text) {
 		else
 			line = p_text.substr(pos, end - pos);
 
-		if (line.length() > 0) {
-
-			if (current->subitems.size() && current->subitems.back()->get()->type == ITEM_TEXT) {
-				//append text condition!
-				ItemText *ti = static_cast<ItemText *>(current->subitems.back()->get());
-				ti->text += line;
-				_invalidate_current_line(main);
-
-			} else {
-				//append item condition
+		int lipos = 0;
+		while (lipos < line.length()) {
+			if (line[lipos] >= 0x3040 && line[lipos] < 0xfaff) {
```

Comment:
This range includes multiple non CJK blocks. Probably should be limited to 3400 — 4DBF, 4E00 — 9FFF, F900 — FAFF and 20000 — 2A6DF, 2F800 — 2FA1F (last two won't work on Windows).

Reply:
For the reference, `master` branch use ICU break iterator with the following rules set:[line_normal_cj.txt](https://raw.githubusercontent.com/unicode-org/icu/master/icu4c/source/data/brkitr/rules/line_normal_cj.txt) and 4MB dictionary: [cjdict.txt](https://raw.githubusercontent.com/unicode-org/icu/master/icu4c/source/data/brkitr/dictionaries/cjdict.txt).

If I understand correctly, this approach should work for pure ideographs, but not for mixed syllabary + ideographs (Okurigana). But since ICU based breaking won't be backported to 3.2, it's probably better than nothing.

Also, I'm not sure if it's good for performance to add a new `ItemText` for each word, it might be better to do it in the `_process_line` instead.